Adds special version variables to docs templating system

andyk opened the pull request:
Adds liquid variables to docs templating system so that they can be used throughout the docs: SPARK_VERSION, SCALA_VERSION, and MESOS_VERSION.

To use them, e.g. use {{site.SPARK_VERSION}}.

Also removes uses of {{HOME_PATH}} which were being resolved to ""
by the templating system anyway.

Commit:
Adds liquid variables to docs templating system so that they can be used
throughout the docs: SPARK_VERSION, SCALA_VERSION, and MESOS_VERSION.

To use them, e.g. use {{site.SPARK_VERSION}}.

Also removes uses of {{HOME_PATH}} which were being resolved to ""
by the templating system anyway.

mateiz commented:
Can you pull the latest change from dev into your branch to make it mergeable? One of the earlier commits made it incompatible.

Commit:
Merge remote-tracking branch 'public-spark/dev' into add-version-vars-to-docs

Conflicts:
	docs/quick-start.md

Commit:
Updates README.md with instructions for running jekyll without building
scaladoc (i.e. run `SKIP_SCALADOC=1 jekyll`).

andyk commented:
Ok, I think I've fixed the conflict. i also updated the README with instructions for building without scaladoc (because this feature can save people lots of time).

mateiz commented:
Great, thanks!
